### DISCUSSION TITLE
Build Zimbra 10.1.0

### DISCUSSION THREAD

**Discussion by MrBoxing** [ianw1974/zimbra-build-scripts#90] [Q&A] [answered]
Hi Ianw, Zimbra just release Zimbra 10.1.0 so how can i build rpm for Zimbra 10.1.0 ?
I had tried to change from "BUILD_RELEASE_NO = 10.0.8" to "BUILD_RELEASE_NO = 10.5.0" in config.build. Zimbra still not release Zimbra 10.5.0 however the build process was still successful. So how can i build rpm for Zimbra 10.1.0. Do you have any advice ? 
Thanks you

![zimbra10-1](https://github.com/user-attachments/assets/5a366660-04b5-40c6-bbfd-c85bf48fb8fe)
![zimbra10-2](https://github.com/user-attachments/assets/302e39d4-f188-4933-8046-a6ec95eced71)

**Comment by MrBoxing** [comment]
@ianw1974 pls help me, thanks you <3

**Reply by jeredfloyd** [reply, marked as answer]
@MrBoxing 10.1 doesn't have any announced security content, so I figure that Ian is sticking to his planned quarterly builds, which means we're about a month out from a Q4 build...

**Reply by ianw1974** [reply]
Exactly as @jeredfloyd mentioned.  I'm not planning to release my next builds until Q4.  My only exceptions for pushing ones out are if there are critical security releases, simply because I do not have the time to continue pushing builds out daily/weekly, etc.  I do also have an issue open to also build Zimbra for later Ubuntu releases which will also be addressed during that next release, since I'll then be able to offer for Ubuntu 22.04 or even 24.04 if Zimbra have also supported for that one as well.

**Reply by MrBoxing** [reply]
As you know, Zimbra Daffodil (v10.1) introduced a new license service with significant changes in licensing management. Can we build an open source edition (v10.1) with GPL license by your script ?

**Comment by ianw1974** [comment]
The way I understand it is that at some point, Zimbra will offer a free license for Zimbra Network Edition that will obviously have certain features restricted, perhaps even a mailbox limit.  At which point, us builders will therefore no longer need to build Zimbra.  We just have to hope and assume that Zimbra then later don't drop the free license.  Of course, the builders could start up again and build the free version again, assuming that they are still interested in doing so at that point.

However, that aside, since my scripts build from the develop branch, even for building 10.0.x pulled in changes and the license service was there.  The caveat was with my builds at that point to **not** install the license service since it's not needed.  Whether that will be the same or not with 10.1, I will find out when I start building and testing at the end of this month.  Assuming it will be the same as it has so far, 10.1 should be up on my site beginning of October.

**Reply by jeredfloyd** [reply]
Of course, now there is announced security content: https://blog.zimbra.com/2024/09/patch-release-multiple-security-issues-related-to-cross-site-scripting-xss-addressed-and-resolved/

These look like all XSS vulnerabilities found via analysis (except the TinyMCE one, but where is this even used?) so I don't think they're high priority.  But if 10.1 ends up being a challenge, I think a latest build on the 10.0.x branch (e.g. 10.0.9) makes sense for your Q4 builds as a fallback.

**Reply by ianw1974** [reply]
According to the link it's just a medium severity.  So yeah, will see how the builds go beginning Q4.

**Comment by ianw1974** [comment]
See this new post here: https://github.com/ianw1974/zimbra-build-scripts/discussions/93 as my builds will be released shortly.

Please also note, going back to the original post, putting ```10.1.0``` or ```10.5.0``` probably doesn't have any effect on the build whatsoever.  It's information you complete, so obviously a build with ```10.5.0``` set will show as being ```10.5.0``` although it's not since that hasn't been released yet.  It's just a number and nothing more although it could have some affect on some things - haven't looked into it, and don't intend to.  Maybe if building from tags this will have more of an influence on whether it fails or not.

As people should know already, my scripts build from develop branch, so it's the latest of whatever is commited to the develop branches at this time - irrespective of whether you put ```10.0.0``` or ```10.1.0``` or even ```9.0.0``` - I do have to make patches for ```9.0.0``` however if things get included from ```10.x``` that shouldn't be there then I have to fix it by patching to enable/disable/remove.
